### PR TITLE
Do not throw UnknownHostException

### PR DESCRIPTION
**Pull request by sgalichkin**
**Original issue description**
Integration test (not available here).
The broker runs in docker container, then goes down and corresponding DNS entry becomes unavailable. After the first attempt to reconnect the `NettyTcpConnection` gets `UnknownHostException` from `NetResolver` and fails to schedule next reconnection attempt.

**Describe your changes**
Modify `NetResolver` to not throw `UnknownHostException`.
This way `NettyTcpConnection` may continue its attempts to (re)connect even if there are temporarily DNS issues (which may disappear in a second).

The proposed solution is not perfect.
Ideally, in case of unknown host it's better to skip connection attempt at all and just schedule the next one but this will require to modify `NettyTcpConnection` (connect/disconnect callbacks) and corresponding tests.

**Testing performed**
Manually tested by starting/stopping broker and modifying `/etc/hosts` records

### CODE REVIEW

File: bmq-sdk/src/test/java/com/bloomberg/bmq/impl/infr/net/NetResolverTest.java [hunk 1]
```diff
@@ -81,10 +82,13 @@ public void testInvalidHosts() {
...
-                // If we reach this line the only resolved host should be localhost
-                assertEquals(1, nr.numResolvedHosts());
-                uri = nr.getNextUri();
-                assertEquals("127.0.0.1", uri.getHost());
+
+                if (nr.isUnknownHost()) {
+                    assertEquals(uri, nr.getNextUri());
+                } else {
+                    // If we reach this line the only resolved host should be localhost
+                    assertEquals("127.0.0.1", nr.getNextUri().getHost());
```
Comment (678098) [inline bmq-sdk/src/test/java/com/bloomberg/bmq/impl/infr/net/NetResolverTest.java:90]: Is it expected that for some invalid hosts we get some valid uri?

Comment (sgalichkin) [reply bmq-sdk/src/test/java/com/bloomberg/bmq/impl/infr/net/NetResolverTest.java:90]: It looks so.
In `NetResolver::init()` the `host` part (if null then `path`) of provided uri is used to get ip addresses.
For uri like `tcp://10000.122.165.92:3011` the host is null and the path is empty. In this case `InetAddress::getAllByName(null|"")` returns `localhost` address.

Do you think this should be fixed?

Comment (sgalichkin) [reply bmq-sdk/src/test/java/com/bloomberg/bmq/impl/infr/net/NetResolverTest.java]: @678098 fixed

Comment (sgalichkin) [reply bmq-sdk/src/test/java/com/bloomberg/bmq/impl/infr/net/NetResolverTest.java]: @678098 fixing the tests.

Comment (sgalichkin) [reply bmq-sdk/src/test/java/com/bloomberg/bmq/impl/infr/net/NetResolverTest.java]: @678098 done

**Pull request merged**